analysers/experiments/IO.py

```diff
@@ -0,0 +1,148 @@
+import sys
+import json
+import io
+import gzip
+import uuid
+import math
+
+import scipy.integrate as integrate
+import scipy.special as special
+import numpy as np
+
+from pyspark_cassandra import CassandraSparkContext
+from pyspark_cassandra import RowFormat
+from pyspark import SparkConf
+
+# Takes arguments: Spark master, Cassandra host, Minio host, path of the files
+sparkMaster = sys.argv[1]
+cassandraHost = sys.argv[2]
+trialID = sys.argv[3]
+experimentID = trialID.split("_")[0]
+cassandraKeyspace = "benchflow"
+srcTable = "trial_io"
+destTable = "exp_io"
+
+# Set configuration for spark context
+conf = SparkConf() \
+    .setAppName("IO analyser") \
+    .setMaster(sparkMaster) \
+    .set("spark.cassandra.connection.host", cassandraHost)
+sc = CassandraSparkContext(conf=conf)
+
+# TODO: Use Spark for all computations
+
+def f1(r):
+    if r['reads'] == None:
+        return (0, 1)
+    else:
+        return (r['reads'], 1)
+    
+def f2(r):
+    if r['writes'] == None:
+        return (0, 1)
+    else:
+        return (r['writes'], 1)
+
+def f3(r):
+    if r['total'] == None:
+        return (0, 1)
+    else:
+        return (r['total'], 1)
+    
+data = sc.cassandraTable(cassandraKeyspace, srcTable)\
+        .select("device") \
+        .where("experiment_id=? AND container_id=?", trialID, experimentID, containerID) \
```

Comment:
You are not using the `trialID` in the where clause, although you set it. I guess you don't need to use the `trialID` since you are computing experiments metrics.